node/src/chain/fsm.rs

```diff
@@ -146,25 +158,31 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> InSyncImpl<DB, VM, N> {
         let iter = acc.get_curr_iteration().await;
         let curr_hash = acc.get_curr_hash().await;
 
-        if h < curr_h {
-            return Ok(false);
-        }
+        info!(
+            event = "fsm::on_event",
+            height = curr_h,
+            iter,
+            blk_height = h,
+            blk_iter = blk.header().iteration,
+            blk_hash = to_str(&blk.header().hash),
+        );
 
-        // Filter out blocks that have already been marked as
-        // blacklisted upon successful fallback execution.
-        if self
-            .blacklisted_blocks
-            .read()
-            .await
-            .contains(&blk.header().hash)
-        {
+        if h < curr_h {
             return Ok(false);
         }
 
         if h == curr_h {
             if blk.header().hash == curr_hash {
                 // Duplicated block.
-                // Node has already accepted it.
+                //
+                // Node has already accepted it (maybe from local consensus)
+                // however we still need to re-propagate it so
+                // that we do not eclipse the network. Eclipsing
+                // the network for that broadcast may damage the
+                // fallback for nodes on different fork.
+
+                self.network.write().await.broadcast(msg).await;
```

Comment:
I'm aware of that. This would be also the case when we implement fallback for any former (non-finalized) block as we'd need to re-broadcast blocks with lower tip.

The reasoning behind this change is: Any node that has accepted a block of 1st iteration will not re-propagate it eclipsing the network for other (non-consensus) guys that are still on the wrong fork.

Reply:
I don't follow the reasoning...
If the block is known, being it because it was received by peers or produced by consensus, it would have been already propagated to the node's peers. 
Re-broadcasting the same block when received from the network would be a duplicate.
If nodes accepted that block at the 1st iteration, they will have broadcasted the same message they are receiving now.

Reply:
Noted. I'll remove it once I confirm that in a case of fallback all nodes are receiving the correct block.